ge/views_utils.py

```diff
@@ -496,10 +497,47 @@ def get_librarydata_results(search_type: str, search_term: str) -> list[LibraryD
 
 def sum_col(ws, col, col_top=5):
     """Add a total row to an Excel spreadsheet column."""
-    col_len = len(ws[col])
+    last_row = get_last_row(ws, col)
+    ws[f"{col}{last_row + 1}"] = f"=SUM({col}{col_top}:{col}{last_row})"
+
+
+def get_last_row(ws, col):
+    """Get the index of the last non-empty row in an Excel spreadsheet column."""
+    last_row = len(ws[col])
     # make sure we get the first non-empty row from the bottom
-    col_len -= next(i for i, x in enumerate(reversed(ws[col])) if x.value is not None)
-    ws[f"{col}{col_len + 1}"] = f"=SUM({col}{col_top}:{col}{col_len})"
+    last_row -= next(i for i, x in enumerate(reversed(ws[col])) if x.value is not None)
+    return last_row
+
+
+def get_last_col(ws, row):
+    """Get the index of the last non-empty column in an Excel spreadsheet row."""
+    last_col = len(ws[row])
+    # make sure we get the first non-empty column from the right
+    last_col -= next(i for i, x in enumerate(reversed(ws[row])) if x.value is not None)
+    return last_col
+
+
+def get_as_of_date() -> str:
```

Comment:
Pass in a date parameter, defaulting to today.  The main benefit: that will allow tests to be written to check correctness for each quarter.